pkg/controller/keyvault/ingress_secret_provider_class.go

```diff
@@ -103,7 +97,18 @@ func (i *IngressSecretProviderClassReconciler) Reconcile(ctx context.Context, re
 		},
 	}
 	logger = logger.WithValues("spc", spc.Name)
-	ok, err := i.buildSPC(ing, spc)
+
+	// Checking if we manage the ingress. All false cases without an error are assumed that we don't manage it
+	ok, err := i.ingressManager.IsManaging(ing)
+
+	if err != nil {
+		return result, fmt.Errorf("determining if ingress is managed: %w", err)
+	}
```

Comment:
to make this more idiomatic we should have the !ok logic here (the cleanup logic).

It doesn't make sense to have multiple if ok { when we can have an early return instead